packages/autorest.go/src/generator/models.ts

```diff
@@ -485,3 +497,153 @@ function generateAliasType(structDef: StructDef, receiver: string, forMarshal: b
   text += `\t}\n`;
   return text;
 }
+
+// represents a method on a model
```

Comment:
Nit, This could be in its own file which is imported.  Would help simplify the amount of code in this file.

Reply:
Given that this type is now exclusively used for models.go codegen, I wanted to keep it local to this file (it was in structs.ts before I renamed it).